drivers/input/input_cf1133.c

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (c) 2023, Kickmaker
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*Address by default is 0x55*/
+
+#define DT_DRV_COMPAT sitronix_cf1133
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(cf1133, CONFIG_INPUT_LOG_LEVEL);
+
+/* cf1133 used registers */
+typedef enum {
+	FIRMWARE_VERSION,
+	STATUS_REG,
+	DEVICE_CONTROL_REG,
+	TIMEOUT_TO_IDLE_REG,
+	XY_RESOLUTION_HIGH,
+	X_RESOLUTION_LOW,
+	Y_RESOLUTION_LOW,
+	DEVICE_CONTROL_REG2 = 0x09,
+	FIRMWARE_REVISION_3 = 0x0C,
+	FIRMWARE_REVISION_2,
+	FIRMWARE_REVISION_1,
+	FIRMWARE_REVISION_0,
+	FINGERS,
+	KEYS_REG,
+	XY0_COORD_H,
+	X0_COORD_L,
+	Y0_COORD_L,
+	I2C_PROTOCOL = 0x3E,
+	MAX_NUM_TOUCHES,
+	DATA_0_HIGH,
+	DATA_0_LOW,
+	MISC_CONTROL = 0xF1,
+	SMART_WAKE_UP_REG = 0xF2,
+	CHIP_ID = 0xF4,
+	PAGE_REG = 0xff,
+} RegisterOffset;
+
+/* Constants  */
+#define SITRONIX_MAX_SUPPORTED_POINT 10U
+
+/* Mask  */
+#define ONE_D_SENSING_CONTROL_SHFT 0x2
+#define ONE_D_SENSING_CONTROL_BMSK 0x3
+#define I2C_PROTOCOL_BMSK          0x3
```

Comment:
you can use  `GENMASK` for these and then access them with `FIELD_GET`, should make things a bit more compact in the code, look around the code base if you need examples

Reply:
Ok done